Library/Homebrew/utils/shared_audits.rb

```diff
@@ -62,9 +62,9 @@ def self.github_release(user, repo, tag, formula: nil, cask: nil)
       [cask.tap&.audit_exception(:github_prerelease_allowlist, cask.token), cask.token, cask.version]
     end
 
-    return "#{tag} is a GitHub pre-release." if release["prerelease"] && [version, "all"].exclude?(exception)
+    return "#{tag} is a GitHub pre-release." if release["prerelease"] && [version, "all", "any"].exclude?(exception)
 
-    if !release["prerelease"] && exception
+    if !release["prerelease"] && exception && exception != "any"
       return "#{tag} is not a GitHub pre-release but '#{name}' is in the GitHub prerelease allowlist."
```

Comment:
It seems weird to insist that the version _must_ be a prerelease if used with `all`. Would getting rid of this produce wrong versions for the casks using it?

Reply:
I agree, it was unexpected to see the audit error when a stable release appeared. Would you prefer I just amend the behaviour of `all` rather than adding a new keyword?

Reply:
Probably, as long as it works fine for the casks that currently use it.

Reply:
It's mainly an error so we don't forget to remove it here once there is a stable release.

Reply:
If I understand correctly, it sounds like there may be a need for an explicit `until-stable` value, that would allow a package to use pre-release versions until a stable release appears. That would allow us to update the behavior for `all` while continuing to handle the aforementioned situation with a new option.

The linked cask PR is a different situation, as upstream unpredictably marks releases using a stable version format as "pre-release" on GitHub, even when their first-party download page links to a GitHub release asset for a "pre-release" version as stable. In that situation we have to either use `all` or use a specific version and accept that almost every version bump will require updating the allowlist to pass CI (increasing maintainer/contributor burden).

Reply:
This was my original intent with splitting the behaviour of `all` and `any` where `all` required *all* releases to be pre-releases (the current behaviour) while `any` allowed *any* releases to be pre-releases.

Reply:
> It's mainly an error so we don't forget to remove it here once there is a stable release, i.e. once it is stable, it should stay stable.

If that's the intent then maybe it would be better to specify versions for those instead of making them `all`.

Reply:
Admittedly this is more complicated and beyond the original scope of this change, but perhaps a range check like `<1.2.0` would be better so you can specify "allow any pre-release up to the stable release 1.2.0"

Reply:
> If that's the intent then maybe it would be better to specify versions for those instead of making them `all`.

Depends on what the more likely case is: I'd guess the case that only pre-releases are published until a stable release is, and not to randomly change between stable/prerelease versions.

The whole point of adding `all` in the first place was to reduce the need to update the allow list with every version bump.

> perhaps a range check

Likely a non-starter because app developers will use pretty much any versioning-scheme except semver. What if `1.1.27` is a pre-release, `1.1.28` is stable, and `1.1.29` is a pre-release again?